sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -128,6 +130,46 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
         ShimLoader.setSparkShimProviderClass(conf.shimsProviderOverride.get)
       }
 
+      // Compare if the cudf version mentioned in the classpath is equal to the version which plugin
+      // expects. If there is a version mismatch, throw error. This check can be disabled by
+      // setting this config spark.rapids.cudfVersionOverride=true
+      if (!conf.cudfVersionOverride) {
+        val props = new Properties
+        val cudfClassLoader = classOf[ai.rapids.cudf.ColumnVector].getClassLoader
+        val cudfProperties = cudfClassLoader.
+          getResourceAsStream("cudf-java-version-info.properties")
+
+        if (cudfProperties == null) {
+          throw new RuntimeException("Could not find the properties file in the cudf jar." +
+            " Cannot verify if the cudf version is same as expected by plugin")
+        }
+        props.load(cudfProperties)
+        if (props.get("version") == null) {
+          throw new RuntimeException("Property name not found in " +
+            "cudf-java-version-info.properties file")
```

Comment:
nit: would be great to have a constant for the name of the properties file.  (Very minor)